lab_6_pipeline/pipeline.py

```diff
@@ -112,11 +198,34 @@ def convert_morphological_tags(self, tags: str) -> str:  # type: ignore
         """
         Converts the Mystem tags into the UD format
         """
+        pos = self.convert_pos(tags)
+        if pos in ["PART", "ADP", "ADV", "CCONJ", "INTJ"]:
+            return ''
+
+        tags = re.sub(r'\((.+?)\|.+\)', r'\1', tags)
+        extracted_tags = re.findall(r'[а-я]+', tags)
+        ud_tags = {}
+        for tag in extracted_tags:
+            for category in (self.animacy, self.case, self.gender, self.number, self.tense):
+                if (pos == 'NOUN' or pos == 'PRON') and category == self.tense:
```

Comment:
this is not a scalable approach. define rules of what attributes to extract for which pos.

Reply:
fixed